### ISSUE TITLE
Usage of multiple print()/data sinks in new Java API

### ISSUE THREAD

**Issue by twalthr** [stratosphere/stratosphere#561]
In the old Java API I could add multiple data sinks to the plan. How do I do this in the new Java API when I just want to use the new print() functionality multiple times? 

Example:

```
final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
env.setDegreeOfParallelism(1);
DataSet<Short> shortDataSet = env.fromElements((short)1,(short)2222,(short)3);      
shortDataSet.filter(new FilterFunction<Short>() {
    @Override
    public boolean filter(Short value) throws Exception {
        return value.equals((short)2222);
    }
}).print();
// second flow
DataSet<Byte> byteDataSet = env.fromElements((byte) 1, (byte) 22, (byte) 3);
byteDataSet.filter(new FilterFunction<Byte>() {
    @Override
    public boolean filter(Byte value) throws Exception {
        return value.equals((byte) 3);
    }
}).print();

env.execute();
```

Results in:

`Exception in thread "main" eu.stratosphere.compiler.CompilerException: The given program contains multiple disconnected data flows.`

**Comment by StephanEwen** [comment]
This is actually on purpose right now, because we assume one program can
only be connected. The same is actually true for the old java api. You can
still use multiple data sinks, the following program should work (two print
sinks):

final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();

DataSet<Short> shortDataSet =
env.fromElements((short)1,(short)2222,(short)3);

shortDataSet.print();

shortDataSet.filter(new FilterFunction<Short>() {
    @Override
    public boolean filter(Short value) throws Exception {
        return value.equals((short)2222);
    }
}).print();

**Comment by twalthr** [comment]
Yes, your example works. But for me (as a new Java API beginner) it would be more intuitive to have the above functionality, too. The old Java API allowed to connect multiple independent flows by adding sinks to the plan.

**Comment by StephanEwen** [comment]
Actually, your example program does not work in the old API as well. You
can add multiple sinks in both the old and new api, but in both only if
they are somehow connected, i.e. have at least one common predecessor.

I would like to support the above case as well. We can do that as soon as
we have reworked the JobManager to accept such job graphs (disconnected job
graphs)
